pkg/cmd/miscellaneous.go

```diff
@@ -344,7 +345,36 @@ func isIP(word string) bool {
 }
 
 func getPublicIP() string {
-	result, err := ExecCmdReturnOutput("bash", "-c", "curl ifconfig.me")
+	result, err := ExecCmdReturnOutput("bash", "-c", "curl icanhazip.com")
 	checkError(err)
 	return result
 }
+
+// get role either user or operator
+func getRole() string {
+	var role string
+	// will refactor in https://github.com/gardener/gardenctl/issues/266
+	KUBECONFIG = getKubeConfigOfClusterType("garden")
+	out, err := ExecCmdReturnOutput("bash", "-c", "kubectl --kubeconfig="+KUBECONFIG+" auth can-i get secret -A")
+	if err == nil {
+		if out == "yes" {
+			role = "operator"
+		}
+	} else {
+		role = "user"
+	}
+	return role
+}
+
+func getTechnicalID() string {
+	var target Target
+	ReadTarget(pathTarget, &target)
+
+	gardenClientset, err := target.GardenerClient()
+	checkError(err)
+	project, err := gardenClientset.CoreV1beta1().Projects().Get(target.Stack()[1].Name, metav1.GetOptions{})
+	checkError(err)
+	shoot, err := gardenClientset.CoreV1beta1().Shoots(*project.Spec.Namespace).Get(target.Stack()[2].Name, metav1.GetOptions{})
+	checkError(err)
+	return string(shoot.Status.TechnicalID)
```

Comment:
```go
return string(shoot.Status.TechnicalID)
```
TechnicalID is already a string, no need to stringify it

Reply:
done